infra/lib/shared/sqs.ts

```diff
@@ -0,0 +1,153 @@
+ *
+ * @param props.createDLQ - create a dead letter queue, default true
+ * @param props.createRetryLambda - create a lambda to retry messages on DLQ, default true
+ * @param props.fifo - whether to create a FIFO queue or not, default false
+ * @returns
+ */
+export function createQueue(props: QueueProps): Queue {
+  const dlq =
+    props.createDLQ === false ? undefined : defaultDLQ(props.stack, props.name, props.fifo);
+  const defaultQueueProps = {
+    ...(dlq ? { dlq: dlq } : {}),
+  };
+  const queue =
+    props.fifo === true
+      ? createFifoQueue({ ...defaultQueueProps, ...props })
+      : createStandardQueue({ ...defaultQueueProps, ...props });
+  props.producer && queue.grantSendMessages(props.producer);
+  props.consumer && queue.grantConsumeMessages(props.consumer);
+  props.consumer && dlq && dlq.grantSendMessages(props.consumer);
+
+  const retryLambda = props.createRetryLambda != null ? props.createRetryLambda : true;
+  if (dlq && retryLambda) {
+    createRetryLambda({
+      ...props,
+      sourceQueue: dlq,
+      destinationQueue: queue,
+    });
+  }
+  return queue;
+}
+
+type AbstractQueueProps = {
+  stack: Construct;
+  name: string;
+  dlq?: Queue;
+  // time a message is invisible to other consumers while its being processed, after this time it will be visible again (can be reprocessed)
+  visibilityTimeout?: Duration;
+  deliveryDelay?: Duration;
+  receiveMessageWaitTime?: Duration;
+  // maximum number of times a message can be processed before being automatically sent to the dead-letter queue
+  maxReceiveCount?: number;
+};
+export type StandardQueueProps = AbstractQueueProps & {
+  contentBasedDeduplication?: never;
+  fifo?: never | false;
+};
+
+function createStandardQueue(props: StandardQueueProps): Queue {
+  return new Queue(props.stack, props.name + "Queue", {
+    queueName: props.name + "Queue",
+    retentionPeriod: Duration.days(14),
+    deliveryDelay: props.deliveryDelay ?? Duration.seconds(0),
+    receiveMessageWaitTime: props.receiveMessageWaitTime ?? Duration.seconds(0),
+    /**
+     * ...set the source queue's visibility timeout to at least six times the timeout that
+     * you configure on your function/lambda:
+     * https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#events-sqs-queueconfig
+     * https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html
+     *
+     * "The extra time allows for Lambda to retry if your function is throttled while processing a previous batch."
+     * This seems to be the case mostly because of the possibility of a lambda being throttled, but in theory we would
+     * only care about this on FIFO queues because we wouldn't want a new lambda to process a message while the original,
+     * throttled lambda is about to keep processing it after its unthrottled.
+     */
+    visibilityTimeout:
+      props.visibilityTimeout ?? Duration.seconds(DEFAULT_LAMBDA_TIMEOUT_SECONDS * 6 + 1),
```

Comment:
nit: the `6` would be good to put in a const at the top of the file since it's used in a couple places